### PR TITLE
Add cmake warning about supported compiler versions.

### PR DESCRIPTION
**Pull request by gnusi**
### Scope & Purpose

Add cmake warning about supported compiler versions.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -604,6 +611,7 @@ if (MSVC)
   endif()
 endif ()
 
+# FIXME: refactor/cleanup check below
```
Comment (markuspf) [inline CMakeLists.txt:614]: Can we not do this now?

Comment (gnusi) [reply CMakeLists.txt:614]: Fore sure, I won't do this now.

Comment (jsteemann) [reply CMakeLists.txt:614]: I think we can remove the previously existing compiler checks safely within this PR, at least the checks for specific compiler versions.
I am also happy if we do it seperately.

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by joerg84 — APPROVED** [review]

**Pull request merged**